Reading List/Reading List/ReadingListTableViewController.swift

```diff
@@ -0,0 +1,113 @@
+        // #warning Incomplete implementation, return the number of rows
+        if section == 0 {
+            return bookController.readBooks.count
+        } else {
+            return bookController.unreadBooks.count
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "BookCell", for: indexPath) as? BookTableViewCell else {
+            return UITableViewCell()}
+        
+        let book = bookFor(indexPath: indexPath)
+        cell.book = book
+        cell.delegate = self
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let book = bookFor(indexPath: indexPath)
+            bookController.deleteBook(with: book)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 0 {
+            if bookController.readBooks.count > 0 {
+                return "Read Books"
+            } else {
+                return nil
+            }
+            
+        } else {
+            if bookController.unreadBooks.count > 0 {
+                return "Unread Books"
+            } else {
+                return nil
+            }
+        }
+    }
+    
+    // MARK: - Navigation
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "AddBookSegue" {
+            guard let bookDetailVC = segue.destination as? BookDetailViewController else {return}
+            bookDetailVC.bookController = bookController
+            bookDetailVC.delegate = self
+        } else {
+            if segue.identifier == "BookDetailSegue" {
+                guard let bookDetailVC = segue.destination as? BookDetailViewController else {return}
+                bookDetailVC.bookController = bookController
+                guard let newIndex = self.tableView.indexPathForSelectedRow else {return}
+                let booked = bookFor(indexPath: newIndex)
+                bookDetailVC.book = booked
+                bookDetailVC.delegate = self
+            }
+        }
+    }
+}
+
+extension ReadingListTableViewController: AddBookDelegate {
+    
+    func bookWasAdded(_ book: Book) {
+        bookController.createBook(with: book.title, reason: book.reasonToRead)
+        tableView.reloadData()
+    }
+    
+    func bookWasUpdated(oldBook: Book, title: String, reason: String) {
+        bookController.updateTitleOrReasson(with: oldBook, title: title, reason: reason)
+        tableView.reloadData()
+    }
+    
+}
```

Comment:
Hell yeah, nice way to let the tableview know about the changes so it can reload the data. Another way would be to use the system method viewWillAppear to call reload data, and you could have the detail view controller communicate with the book controller directly, as you passed a reference when segueing.